### PR TITLE
CREATED help README file to add missing details

### PR DESCRIPTION
**Pull request by bryantson**
Running the Template file needs the additional information about how to get the required arguments

#### What is this PR About?
Deploying the Prometheus documentation is confusing as it misses the information about the required parameters.

#### How do we test this?
1. Create  a new project in Openshift using "oc new-project test-prometheus"
2. Deploy as a new app using either
 oc new-app -f template.yaml 

or 
oc process -f template.yaml  > resource.yaml
then 
oc create -f resource.yaml

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Thank you @bryantson 
A few comments, I like the idea of documenting the template parameters. Typically I'd use the `description` field of the parameter for this.
If we want to use a README I think it should include all the template parameters (including the optional).
Maybe use a [table](https://www.markdownguide.org/extended-syntax/#tables) to describe the parameters, required/optional, default values and description.

File: prometheus/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+# Deploys the Prometheus using template
+
+### REQUIRED ARGUMENTS -- Need more details
+
+
+**1. NAMESPACE: Use to override the default namespace created under project**
+- More description are needed
```
Comment (pabrahamsson) [inline prometheus/README.md:8]: Either delete this line or put something more descriptive here.

**Comment by pabrahamsson** [comment]
@bryantson will you be able to update this PR?

**Comment by bryantson** [comment]
@pabrahamsson Sorry. I just read this. Sure.

**Comment by garethahealy** [comment]
@bryantson ; if you are going to duplicate the content from the template into the readme, can you try to use some type of tool which can copy/process the yaml into the readme?

I am thinking of something like:
- https://github.com/zakhenry/embedme

It's mainly to stop the readme and template going out of sync.

@pabrahamsson ; thoughts?

**Comment by pabrahamsson** [comment]
@garethahealy not sure using embedme would be suitable here. I think all we're looking for is to document the template parameters (which again can be done in the `description` field of the template). My understanding is that embedme would include the whole template file, correct?
@bryantson will you be able to follow up with your intentions here? Are you looking to create a README or are you ok with updating the `description` filed of the parameters?

**Comment by garethahealy** [comment]
@pabrahamsson just to clarify. i agree with you that the "source of truth" for the parameters should be the template file and not something else. so if this PR is just to add more detail to the descriptions, then the embedme point can be ignored.

however, if @bryantson was going to include the params in the readme (which it seems hes started to do), then i think it would be a good idea to look into using a tool which could automate that. embedme was an example, which can select a line number range from a file and insert it into an md file. its probably not the final answer, as am sure there are tools which could process the yaml template and convert the params to an md-table which would be nicer.

does that make sense?

**Comment by pabrahamsson** [comment]
Thank you for the clarification @garethahealy, I had missed the line number range option which certainly makes this tool quite useful. I think you and I are both on the same page here and at this point I'd like to hear from @bryantson and what his thoughts are.

**Comment by bryantson** [comment]
Need to get back to you. I need to try this to ensure what to recommend.

**Comment by garethahealy** [comment]
@bryantson ; any update on this? just to see if we can progress it/get it merged.

**Comment by garethahealy** [comment]
closing due to inactivity

**Pull request closed without merging**